### ISSUE TITLE
"E3035 DeletionPolicy should be only one of Delete, Retain, Snapshot" need addition of 

### ISSUE THREAD

**Issue by michft-v** [aws-cloudformation/cfn-lint#2861]
### CloudFormation Lint Version

cfn-lint 0.79.7

### What operating system are you using?

MasOS 13.5.1

### Describe the bug

E3035 finds "DeletionPolicy: RetainExceptOnCreate" as an error but it should be allowed as it has been added to Cloudformation.

https://aws.amazon.com/about-aws/whats-new/2023/07/aws-cloudformation-deletion-policies-dev-test-cycle/
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html

### Expected behavior

I expected to not error as RetainExceptOnCreate is currently a valid DeletionPolicy: value.

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: 2010-09-09


Parameters:

  AccessSubnetCidrs:
    Type: CommaDelimitedList

  EnterpriseVpcId:
    Type: String

Resources:
  AZaAccessSubnet:
    Properties:
      AvailabilityZone:
        Fn::Select:
        - 0
        - Fn::GetAZs: ''
      CidrBlock: !Select [0, !Ref AccessSubnetCidrs]
      VpcId: !Ref EnterpriseVpcId
    DeletionPolicy: RetainExceptOnCreate
    UpdateReplacePolicy: Retain
    Type: AWS::EC2::Subnet

```

**Comment by kddejong** [comment]
You sure you are on `v0.79.7`?
This was fixed in #2834  which was included in the v0.79.7 release.

Willing to reopen as needed.
